clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
Yeah, good point. We could probably fix this by adding the resetGeneration call to KafkaConsumer.assign as well. Maybe that's the only place that needs it? In general, it's feeling kind of nasty having to keep SubscriptionState synchronized with the internal Coordinator state.